src/interface/variable.hpp

```diff
@@ -204,6 +204,40 @@ class EdgeVariable {
   std::string label_;
 };
 
+template <typename T>
+class ParticleVariable {
+ public:
+  /// Initialize a particle variable
+  ParticleVariable(const std::string label, const int npool, const Metadata &metadata)
+      : data(label, npool), npool_(npool), m_(metadata), label_(label) {}
+
+  // accessors
+  KOKKOS_FORCEINLINE_FUNCTION
+  ParArrayND<T> &Get() { return data; }
```

Comment:
Make method noexcept. Again you are returning a non const reference, this will break encapsulation.

Reply:
We want to modify `data` directly, `ParticleVariable` is just a container for this, right? `data` is a public variable here and in the other `*Variable` classes. It looks like the other classes don't have `Get()` methods, I think they might have when I originally wrote this

Reply:
Ok, if that's the true, then this change goes beyond the scope of your pull request.